account_financial_report/report/journal_ledger.py

```diff
@@ -209,36 +218,34 @@ def _get_move_lines(self, move_ids, wizard, journal_ids):
                     "description": tax_description,
                 }
         Move_Lines = {}
-        accounts = self.env["account.account"]
-        partners = self.env["res.partner"]
-        currencies = self.env["res.currency"]
-        tax_lines = self.env["account.tax"]
         auto_sequence = len(move_ids)
+        Move_Lines = defaultdict(list)
+
         for ml in move_lines:
-            if ml.account_id not in accounts:
-                accounts |= ml.account_id
-            if ml.partner_id not in partners:
-                partners |= ml.partner_id
-            if ml.currency_id not in currencies:
-                currencies |= ml.currency_id
-            if ml.tax_line_id not in tax_lines:
-                tax_lines |= ml.tax_line_id
-            if ml.move_id.id not in Move_Lines.keys():
-                Move_Lines[ml.move_id.id] = []
+            move_id = ml.move_id.id
+            if move_id not in Move_Lines:
                 auto_sequence -= 1
-            taxes = (
-                ml.id in move_line_ids_taxes_data.keys()
-                and move_line_ids_taxes_data[ml.id]
-                or {}
-            )
-            exigible = ml in move_lines_exigible
-            Move_Lines[ml.move_id.id].append(
+            taxes = move_line_ids_taxes_data.get(ml.id, {})
+            exigible = ml.id in move_lines_exigible_ids
+            Move_Lines[move_id].append(
                 self._get_move_lines_data(ml, wizard, taxes, auto_sequence, exigible)
             )
-        account_ids_data = self._get_account_data(accounts)
-        partner_ids_data = self._get_partner_data(partners)
-        currency_ids_data = self._get_currency_data(currencies)
-        tax_line_ids_data = self._get_tax_line_data(tax_lines)
+        account_ids_data = self._get_account_data(
+            move_lines.mapped("account_id").with_context(
```

Comment:
No need of the `mapped` (just syntactic sugar for better readability):

```suggestion
            move_lines.account_id.with_context(
```

Reply:
Changed